src/app/clusters/software-diagnostics-server/CodegenIntegration.cpp

```diff
@@ -18,31 +18,41 @@
 #include <app/clusters/software-diagnostics-server/software-diagnostics-logic.h>
 #include <app/static-cluster-config/SoftwareDiagnostics.h>
 #include <data-model-providers/codegen/CodegenDataModelProvider.h>
+#include <lib/core/Global.h>
 
 using namespace chip;
 using namespace chip::app;
 using namespace chip::app::Clusters;
 using namespace chip::app::Clusters::SoftwareDiagnostics;
+using chip::app::Clusters::SoftwareDiagnostics::StaticApplicationConfig::IsAttributeEnabledOnSomeEndpoint;
 
 // this file is ever only included IF software diagnostics is enabled and that MUST happen only on endpoint 0
 static_assert(SoftwareDiagnostics::StaticApplicationConfig::kFixedClusterConfig.size() == 1,
-              "Exactly one softare diagnistics provider may exist on Endpoint 0");
+              "Exactly one software diagnostics cluster instance may exist");
 static_assert(SoftwareDiagnostics::StaticApplicationConfig::kFixedClusterConfig[0].endpointNumber == kRootEndpointId,
-              "Exactly one softare diagnistics provider may exist on Endpoint 0");
+              "The software diagnostics cluster must be on endpoint 0");
 
 namespace {
 
-RegisteredServerCluster<SoftwareDiagnosticsServerCluster<DeviceLayerSoftwareDiagnosticsLogic>> gServer;
+Global<LazyRegisteredServerCluster<SoftwareDiagnosticsServerCluster<DeviceLayerSoftwareDiagnosticsLogic>>> gServer;
 
 } // namespace
 
 void MatterSoftwareDiagnosticsPluginServerInitCallback()
 {
     // NOTE: we assume code-generation logic is always correct here (we assert at least kFixedClusterConfig settings)
     //       so no error checks are done.
-    (void) CodegenDataModelProvider::Instance().Registry().Register(gServer.Registration());
+    const SoftwareDiagnosticsEnabledAttributes enabledAttributes{
+        .enableThreadMetrics     = IsAttributeEnabledOnSomeEndpoint(Attributes::ThreadMetrics::Id),
```

Comment:
This seems a bit weird, in the sense that what we really care about here are which attributes are enabled on endpoint 0, no?  That's where we're initializing our cluster...

Reply:
We assert that we have exactly on attribute on endpoint 0. I could use the endpoint0 constants, however I was wondering if those would constexpr-out or if they would pull in extra metadata. For now this implementation is more clearly constexpr-out.

Is this about code clarity or do you thing it does something wrong?

Reply:
Updated code to use the "exactly enabled on endpoint 0". These should be equivalend and a flash diff says the new method is 10 bytes of flash longer, however that seems ok for readability as we are still in the 200 bytes overhead ballpark.